Content.Server/Weapon/Melee/EnergySword/EnergySwordSystem.cs

```diff
@@ -0,0 +1,152 @@
+        }
+
+        private void OnComponentInit(EntityUid uid, EnergySwordComponent comp, ComponentInit args)
+        {
+            if (comp.ColorOptions.Count != 0)
+                comp.BladeColor = _random.Pick(comp.ColorOptions);
+        }
+
+        private void OnMeleeHit(EntityUid uid, EnergySwordComponent comp, MeleeHitEvent args)
+        {
+            
+            if (comp.Activated == true)
+            {
+                args.BonusDamage.DamageDict = new Dictionary<string, FixedPoint2>()
+                {
+                    {"Slash", FixedPoint2.New(25)},
+                };
+                args.HitSoundOverride = comp.HitSound;
+            }
+        }
+
+        private void OnUseInHand(EntityUid uid, EnergySwordComponent comp, UseInHandEvent args)
+        {
+            if (!_blockerSystem.CanUse(args.User))
+                return;
+
+            if (comp.Activated)
+            {
+                TurnOff(comp);
+            }
+            else
+            {
+                TurnOn(comp, args.User);
+            }
+        }
+
+        private void TurnOff(EnergySwordComponent comp)
+        {
+            if (!comp.Activated)
+                return;
+
+            if (!TryComp(comp.Owner, out SpriteComponent? sprite)
+                || !TryComp(comp.Owner, out SharedItemComponent? item)
+                || !TryComp(comp.Owner, out SharedPointLightComponent? light))
+                return;
+
+            item.Size = 5;
+
+            SoundSystem.Play(Filter.Pvs(comp.Owner), comp.DeActivateSound.GetSound(), comp.Owner);
+
+            item.EquippedPrefix = "off";
+
+            if (comp.Hacked == true)
+            {
+                sprite.LayerSetState(0, "e_sword");
+            }
+            else
+            {
+                sprite.LayerSetVisible(1, false);
+            }
+
+            light.Enabled = false;
+            comp.Activated = false;
+        }
+
+        private void TurnOn(EnergySwordComponent comp, EntityUid user)
+        {
+            if (comp.Activated)
+                return;
+
+            if (!TryComp(comp.Owner, out SpriteComponent? sprite)
+                || !TryComp(comp.Owner, out SharedItemComponent? item)
+                || !TryComp(comp.Owner, out SharedPointLightComponent? light))
+                return;
+
+            item.Size = 9999;
+
+            SoundSystem.Play(Filter.Pvs(comp.Owner), comp.ActivateSound.GetSound(), comp.Owner);
+
+            if (comp.Hacked == true)
+            {
+                item.EquippedPrefix = "on-rainbow";
+                sprite.LayerSetState(0, "e_sword_rainbow_on");
+            }
+            else
+            {
+                item.EquippedPrefix = "on";
+                item.Color = comp.BladeColor;
+
+                sprite.LayerSetColor(1, comp.BladeColor);
+                sprite.LayerSetVisible(1, true);
+
+                light.Color = comp.BladeColor;
+                light.Enabled = true;
+            }
```

Comment:
This should really be done on the client as all visualizers are:
- Lower bandwidth
- Actually works with photography

Reply:
Got it! I'll figure out how to do that and re-request review after it's done, thanks!